### DISCUSSION TITLE
Avoid additional Mongock Configuration

### DISCUSSION THREAD

**Discussion by WLefever-Cegeka** [Informatievlaanderen/VSDS-LDESServer4J#741] [Ideas]
Current version of the server requires mongock configuration to start.
Another option is to hardwire that configuration in a bean, e.g. always execute all mongock changesets.
That would avoid that users need to add additional configuration to their application.yml.

https://docs.mongock.io/v5/runner/springboot/index.html

**Comment by Tomvbe** [comment]
I think this is risky since our mongock changesets are often not backwards compatible. If you run changeset 1 on a more recent database, it will break the data. My main concern is people running different states of version 0 at the moment. If we they started using the server without mongock but have the datamodel of changeset 3. Suddenly running all unrun changesets on their server would mess up the data. If we do implement this, we should perhaps do this as of v1.

**Reply by Tomvbe** [reply]
An alternative would be the following setup:

Allow config of changesets on an individual level. This can be useful for example if you are only interested in changeset7 below which migrates your yaml config to database config.

```
mongock:
  migration-scan-package:
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset1
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset2
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset3
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset4
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset5
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset6
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0.changeset7
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v1.changeset8
```

Since this list will get quite long, I would propose grouping them by major release. Notice the v0 and v1 I have added in the package structure. This would make it easier for users to upgrade to new versions, they would just need one more package.

```
mongock:
  migration-scan-package:
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v0
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock.v1
```

If you want to run all changesets you can just add:
```
mongock:
  migration-scan-package:
    - be.vlaanderen.informatievlaanderen.ldes.server.infra.mongo.mongock
```

If you want no changesets to run, you add no configuration.
